test-data/unit/check-serialize.test

```diff
@@ -0,0 +1,1202 @@
+-- Serialization test cases (incremental type checking)
```

Comment:
Where did you get all these tests? I hope you don't mind I'm not going to review all of them carefully, I trust you.